tremor-script/src/ast/raw.rs

```diff
@@ -137,7 +139,25 @@ impl<'script> ScriptRaw<'script> {
                 .map(|d| d.iter().map(|l| l.trim()).collect::<Vec<_>>().join("\n")),
         });
 
+        let start = Location {
```

Comment:
We could derive Default for Location that way we could use Location::default() here and below and also avoid the extra allow

Reply:
Yes - Q though - would we use the actual file/stream/buffer offsets by line/column/byte or something that's identifiable as a marker eg: -1/-1/-1 which would mean 'start of stream' or 'end of stream' respectively. If the latter, perhaps -1/-1/-1 is a sensible default here for Script/Query/Deploy - but it wouldn't be for any other AST node that must be within a file/stream/buffer ... ?

Reply:
For embedded queries - location would have a real line/col/byte offset - so that's nice i think

Reply:
An enum perhaps? That avoids the magic numbers and later confusion how we're at -`1 byte (I know I'll be confused by that :joy: eventually)

` ` ` 
enum Location {
  Start,
  End,
  Position {...}
}
` ` `

Reply:
Enum might be the one yes